presto-main/src/main/java/com/facebook/presto/server/StatementResource.java

```diff
@@ -118,10 +132,14 @@
 {
     private static final Logger log = Logger.get(StatementResource.class);
 
-    private static final Duration MAX_WAIT_TIME = new Duration(1, SECONDS);
+    private static final Duration DEFAULT_WAIT_TIME = new Duration(1, SECONDS);
+    private static final Duration MAX_WAIT_TIME = new Duration(10, SECONDS);
```

Comment:
We record one heartbeat per call, so max wait time must be much less than heartbeat check duration (`com.facebook.presto.execution.QueryManagerConfig#clientTimeout`, min value `5s`).